test/sequential/rolldown.withdraw.fees.test.ts

```diff
@@ -0,0 +1,164 @@
+  );
+  const events = await signTx(
+    getApi(),
+    await Withdraw(testUser, withdrawalAmount, ethIdL1Asset.ethereum, chain),
+    testUser.keyRingPair,
+  );
+  const eventFiltered = filterZeroEventData(events, "Transfer");
+  const transferAmountBefore = eventFiltered.amount.replaceAll(",", "");
+  const transferAmount = new BN(transferAmountBefore);
+  expect(eventFiltered.from).toEqual(testUser.keyRingPair.address);
+  expect(eventFiltered.to).toEqual(treasuryAccount);
+  const treasuryBalanceAfter = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  expect(treasuryBalanceAfter.free).bnEqual(
+    treasuryBalanceBefore.free.add(transferAmount),
+  );
+});
+
+test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing ALL GASP, extrinsic fail", async () => {
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+  const tokenAmount = await api.query.tokens.accounts(
+    testUser.keyRingPair.address,
+    GASP_ASSET_ID,
+  );
+  await signTx(
+    getApi(),
+    await Withdraw(testUser, tokenAmount.free, gaspIdL1Asset.ethereum, chain),
+    testUser.keyRingPair,
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("NotEnoughAssets");
+  });
+});
+
+test("GIVEN a withdrawal, WHEN user having only Eth and withdrawing Eth, extrinsic fail", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(ETH_ASSET_ID, testUser, DEFAULT_AMOUNT),
+  );
+  await signTx(
+    getApi(),
+    await Withdraw(testUser, withdrawalAmount, ethIdL1Asset.ethereum, chain),
+    testUser.keyRingPair,
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("NotEnoughAssetsForFee");
+  });
+});
+
+test("Given a fee withdrawal payment, tokens go to Treasury", async () => {
+  const [testUser2] = setupUsers();
+  const treasuryBalanceBefore = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser2));
+  const events = await signTx(
+    getApi(),
+    await Withdraw(testUser, withdrawalAmount, gaspIdL1Asset.ethereum, chain),
+    testUser2.keyRingPair,
+  );
+  const transferEvent = filterZeroEventData(events, "Transfer");
+  const transferAmountBefore = transferEvent.amount.replaceAll(",", "");
+  const transferAmount = new BN(transferAmountBefore);
+  expect(transferEvent.from).toEqual(testUser2.keyRingPair.address);
+  expect(transferEvent.to).toEqual(treasuryAccount);
+  const treasuryBalanceAfter = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  expect(treasuryBalanceAfter.free).bnEqual(
+    treasuryBalanceBefore.free.add(transferAmount),
+  );
```

Comment:
is this not duplicated? The only thing that changes is the dest addrr,

Reply:
Here we check that we can make a withdrawal when our account is not a recipient.

Reply:
lets keep it then , but rename it accordingly

Reply:
Renamed

Reply:
Renamed